src/D2L.CodeStyle.Analyzers/Immutability/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -110,5 +119,12 @@ INamedTypeSymbol analyzedType
 
 			return false;
 		}
+
+		private static bool IsFromGeneratedResourcesFile( ISymbol symbol ) {
+			var tree = symbol.DeclaringSyntaxReferences.Single().SyntaxTree;
+
+			return tree.FilePath.EndsWith( ".Designer.cs" )
+				&& tree.ToString().StartsWith( "//------------------------------------------------------------------------------\n// <auto-generated>" );

```

Comment:
Just this marker text is enough to exclude the file in `UnsafeStaticsAnalyzer` right now. Restricting it to just `*.Designer.cs` so we analyze more things at least.